src/synthesizer/load_data/load_bluetides.py

```diff
@@ -0,0 +1,320 @@
+            pig.open("5/BlackholeMass")[0:length_of_bhar][
+                self.luminous_BH_indices
+            ]
+            * 1e10
+            / self.hh
+        )  # masses of BHs with largest accretion rate
+
+        self.bhid = pig.open("5/ID")[0:length_of_bhar][
+            self.luminous_BH_indices
+        ]
+
+        self.bh_position = (
+            np.transpose(
+                pig.open("5/Position")[0:length_of_bhar][
+                    self.luminous_BH_indices
+                ]
+            )
+            / self.hh
+            / (1 + self.z)
+        )
+
+        self.bhar = self.bhar[self.luminous_BH_indices]
+
+        # Set particle location indices for the BHs in order in pig
+        offset_index = []  # index in offset
+        for i in self.luminous_BH_indices:
+            offset_index.append(get_placed(i, self.bh_offset))
+        self.offsetIndex = np.array(offset_index)
+
+        # Load stellar particle information
+        self.metallicity_ind = pig.open("4/Metallicity")[
+            0 : StarOffset[offset_index[np.argmax(StarOffset[offset_index])]]
+        ]
+        self.star_formation_time_ind = pig.open("4/StarFormationTime")[
+            0 : StarOffset[offset_index[np.argmax(StarOffset[offset_index])]]
+        ]
+        self.position_ind = (
+            pig.open("4/Position")[
+                0 : StarOffset[
+                    offset_index[np.argmax(StarOffset[offset_index])]
+                ]
+            ]
+            / self.hh
+            / (1 + self.z)
+        )
+        self.velocity_ind = (
+            pig.open("4/Velocity")[
+                0 : StarOffset[
+                    offset_index[np.argmax(StarOffset[offset_index])]
+                ]
+            ]
+            / self.hh
+        )
+        self.met_surface_densities_ind = sunset.open("4/MetSurfaceDensity")[
+            0 : StarOffset[offset_index[np.argmax(StarOffset[offset_index])]]
+        ]
+
+
+def load_BlueTides(
+    redshift,
+    dataholder=None,
+    galaxy_bhid=[],
+    end_arr=108001,
+    bluetides_data_folder="",
+):
+    """
+    Load BlueTides galaxies into a galaxy object
+
+    Args:
+        redshift (float):
+            desired redshift to extract BlueTides galaxies
+        dataholder (BlueTidesDataHolder object):
+            contains all data for a particular simulation of BlueTides
+            (default is `None` and dataholder class will be generated from
+            a BlueTides simulation file for the given redshift)
+        galaxy_bhid (array):
+            contains all the galaxy BHIDs to include in your galaxy object
+            Note: each BlueTides dataholder has arrays sorted in descending
+            bhar (default is an empty list)
+        end_arr (int):
+            last galaxy to include in BlueTides dataholder, selection of this
+            value should be done with care to avoid galaxies with BHs too close
+            to the seed mass (10^5.8 solar masses; default is 108001)
```

Comment:
The total number of haloes and star particles varies by redshift in BlueTides. If we try and load in everything for each redshift, we usually run out of RAM. BlueTides is roughly sorted by star particle count within the halo. My idea was that if you're using BlueTides, you'll probably already know approximately how many haloes actually contain black holes far enough from the seed mass that you'll include them? So that's the reason for the strange magic number... but I agree it's a bit weird. I have provided approximate values to include for redshifts I've looked at.

Reply:
I think this is fine for now, but I think it might be better to provide more flexible arguments enabling filtering the loaded galaxies / haloes by mass